src/base/tool-manager/tool-manager.ts

```diff
@@ -0,0 +1,143 @@
+/*******************************************************************************
+ * Copyright (c) 2018 EclipseSource
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * 	Philip Langer - initial API and implementation
+ ******************************************************************************/
+import { inject, injectable } from "inversify";
+import { TYPES } from "../types";
+import { EnableStandardToolsAction, EnableToolsAction, Tool } from "./tool";
+import { IActionHandlerInitializer, IActionHandler, ActionHandlerRegistry } from "../actions/action-handler";
+import { Action } from "../actions/action";
+import { ICommand } from "../commands/command";
+import { KeyListener } from "../views/key-tool";
+import { SModelElement } from "../model/smodel";
+import { matchesKeystroke } from "../../utils/keyboard";
+
+/**
+ * A tool manager coordinates the state of tools in the context of an editor.
+ *
+ * One instance of a tool manager is intended per editor, coordinating the state of all tools within
+ * this editor. A tool can be active or not. A tool manager ensures that activating a set of tools
+ * will disable all other tools, allowing them to invoke behavior when they become enabled or disabled.
+ */
+export interface ToolManager {
+
+    /** All tools managed by this tool manager. */
+    readonly managedTools: Tool[];
+
+    /** The tools that are enabled by standard, whenever no other tool is enabled. */
+    readonly standardTools: Tool[];
```

Comment:
Yes, I agree. I think I originally avoided the term default to distinguish it from the `DefaultToolManager` implementation, but since we should put an `I` prefix before the interface anyway, we avoid the name clash and can call the default implementation simply `ToolManager`.